spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/ReadConf.scala

```diff
@@ -20,15 +21,18 @@ object ReadConf {
   val DefaultSplitSize = 100000
   val DefaultFetchSize = 1000
   val DefaultConsistencyLevel = ConsistencyLevel.LOCAL_ONE
+  import CassandraConnectorConf.convert
+  def fromSparkConf(conf: SparkConf): ReadConf = getReadConf(conf)
 
-  def fromSparkConf(conf: SparkConf): ReadConf = {
+  def fromSparkConf(conf: SparkConf, cluster: Option[String] = None): ReadConf = getReadConf(conf, cluster)
+
+  private def getReadConf(conf: SparkConf, cluster: Option[String] = None): ReadConf = {
     ReadConf(
-      fetchSize = conf.getInt("spark.cassandra.input.page.row.size", DefaultFetchSize),
-      splitSize = conf.getInt("spark.cassandra.input.split.size", DefaultSplitSize),
+      fetchSize = conf.getInt(convert("spark.cassandra.input.page.row.size", cluster), DefaultFetchSize),
```

Comment:
It's required for Spark SQL join, also I can image may developer like to set one single SparkConf instead of each time create new ReadConf for each single statement.